File: python/estruturado/consumidor_estruturado.py
# Dicionários para armazenar dados dos consumidores
consumidores = []

def cadastrar_consumidor(nome, endereco, cidade):
    """Cadastra um novo consumidor"""
    consumidor = {
        'nome': nome,
        'endereco': endereco,
        'cidade': cidade
    }
    consumidores.append(consumidor)
    print("✓ Consumidor cadastrado com sucesso!")

def listar_consumidores():
    """Lista todos os consumidores cadastrados"""
    if not consumidores:
        print("Nenhum consumidor cadastrado.")
        return
    
    print("\n=== LISTA DE CONSUMIDORES ===")
    for i, c in enumerate(consumidores, 1):
        print(f"{i}. Nome: {c['nome']} | Endereco: {c['endereco']} | Cidade: {c['cidade']}")
    print("=============================\n")

def buscar_consumidor(nome):
    """Busca um consumidor por nome"""
    for c in consumidores:
        if c['nome'].lower() == nome.lower():
            print(f"\n✓ Consumidor encontrado!")
            print(f"Nome: {c['nome']} | Endereco: {c['endereco']} | Cidade: {c['cidade']}\n")
            return
    print("\n✗ Consumidor não encontrado.\n")

def deletar_consumidor(nome):
    """Deleta um consumidor por nome"""
    for i, c in enumerate(consumidores):
        if c['nome'].lower() == nome.lower():
            consumidores.pop(i)
            print("✓ Consumidor removido com sucesso!")
            return
    print("✗ Consumidor não encontrado.")

def menu():
    """Menu interativo do sistema"""
    while True:
        print("\n=== SISTEMA DE CADASTRO DE CONSUMIDORES ===")
        print("1. Cadastrar consumidor")
        print("2. Listar consumidores")
        print("3. Buscar consumidor")
        print("4. Deletar consumidor")
        print("5. Sair")
        
        opcao = input("Escolha uma opção: ")
        
        if opcao == '1':
            nome = input("Nome: ")
            endereco = input("Endereco: ")
            cidade = input("Cidade: ")
            cadastrar_consumidor(nome, endereco, cidade)
        
        elif opcao == '2':
            listar_consumidores()
        
        elif opcao == '3':
            busca = input("Digite o nome para buscar: ")
            buscar_consumidor(busca)
        
        elif opcao == '4':
            deletar = input("Digite o nome para deletar: ")
            deletar_consumidor(deletar)
        
        elif opcao == '5':
            print("Encerrando...")
            break
        
        else:
            print("Opção inválida!")

if __name__ == "__main__":
    menu()

File: java/estruturado/ConsumidorEstruturado.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ConsumidorEstruturado {
    
    // Estrutura de dados para armazenar consumidores
    static class Consumidor {
        String nome;
        String endereco;
        String cidade;
    }
    
    static List<Consumidor> consumidores = new ArrayList<>();
    static Scanner scanner = new Scanner(System.in);
    
    // Função para cadastrar um consumidor
    static void cadastrarConsumidor(String nome, String endereco, String cidade) {
        Consumidor c = new Consumidor();
        c.nome = nome;
        c.endereco = endereco;
        c.cidade = cidade;
        consumidores.add(c);
        System.out.println("✓ Consumidor cadastrado com sucesso!");
    }
    
    // Função para listar todos os consumidores
    static void listarConsumidores() {
        if (consumidores.isEmpty()) {
            System.out.println("Nenhum consumidor cadastrado.");
            return;
        }
        System.out.println("\n=== LISTA DE CONSUMIDORES ===");
        for (int i = 0; i < consumidores.size(); i++) {
            Consumidor c = consumidores.get(i);
            System.out.println((i + 1) + ". Nome: " + c.nome + " | Endereco: " + c.endereco + " | Cidade: " + c.cidade);
        }
        System.out.println("=============================\n");
    }
    
    // Função para buscar consumidor por nome
    static void buscarConsumidor(String nome) {
        for (Consumidor c : consumidores) {
            if (c.nome.equalsIgnoreCase(nome)) {
                System.out.println("\n✓ Consumidor encontrado!");
                System.out.println("Nome: " + c.nome + " | Endereco: " + c.endereco + " | Cidade: " + c.cidade + "\n");
                return;
            }
        }
        System.out.println("\n✗ Consumidor não encontrado.\n");
    }
    
    // Função para deletar consumidor
    static void deletarConsumidor(String nome) {
        for (int i = 0; i < consumidores.size(); i++) {
            if (consumidores.get(i).nome.equalsIgnoreCase(nome)) {
                consumidores.remove(i);
                System.out.println("✓ Consumidor removido com sucesso!");
                return;
            }
        }
        System.out.println("✗ Consumidor não encontrado.");
    }
    
    // Menu interativo
    static void menu() {
        while (true) {
            System.out.println("\n=== SISTEMA DE CADASTRO DE CONSUMIDORES ===");
            System.out.println("1. Cadastrar consumidor");
            System.out.println("2. Listar consumidores");
            System.out.println("3. Buscar consumidor");
            System.out.println("4. Deletar consumidor");
            System.out.println("5. Sair");
            System.out.print("Escolha uma opção: ");
            
            int opcao = scanner.nextInt();
            scanner.nextLine();
            
            switch (opcao) {
                case 1:
                    System.out.print("Nome: ");
                    String nome = scanner.nextLine();
                    System.out.print("Endereco: ");
                    String endereco = scanner.nextLine();
                    System.out.print("Cidade: ");
                    String cidade = scanner.nextLine();
                    cadastrarConsumidor(nome, endereco, cidade);
                    break;
                case 2:
                    listarConsumidores();
                    break;
                case 3:
                    System.out.print("Digite o nome para buscar: ");
                    String busca = scanner.nextLine();
                    buscarConsumidor(busca);
                    break;
                case 4:
                    System.out.print("Digite o nome para deletar: ");
                    String deletar = scanner.nextLine();
                    deletarConsumidor(deletar);
                    break;
                case 5:
                    System.out.println("Encerrando...");
                    scanner.close();
                    return;
                default:
                    System.out.println("Opção inválida!");
            }
        }
    }
    
    public static void main(String[] args) {
        menu();
    }
}